### PR TITLE
build(deps): upgrade to `go1.18.3`

### PR DESCRIPTION
**Pull request by matt-boris**
> Upgrade to `go1.18.3` from `go1.18.2`.

### CODE REVIEW

**Comment by matt-boris** [comment]
Rebasing branch off of https://github.com/taskcluster/taskcluster/pull/5496 to get the `lint-golang` task fix. So https://github.com/taskcluster/taskcluster/pull/5496 should be merged before this one.

File: db/test-setup.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#! /bin/bash
+#!/bin/bash
```
Comment (lotas) [inline db/test-setup.sh:1]: ouch! nice one

Comment (matt-boris) [reply db/test-setup.sh:1]: First three commits on this PR are from https://github.com/taskcluster/taskcluster/pull/5496 btw :)

Comment (lotas) [reply db/test-setup.sh:1]: hmm, actually space is allowed :D https://en.wikipedia.org/wiki/Shebang_(Unix)#History  who knew

Comment (lotas) [reply db/test-setup.sh:1]: > First three commits on this PR are from #5496 btw :)

Approving both

**Review by lotas — APPROVED** [review]
Thanks!

**Pull request merged**